scripts/sysreadyshow

```diff
@@ -0,0 +1,120 @@
+header_detail = ['Service-Name', 'Service-Status', 'App-Ready-Status', 'Down-Reason', 'AppStatus-UpdateTime']
+
+SERVICE_STATUS_TABLE = 'ALL_SERVICE_STATUS'
+SYSREADY_TABLE = "SYSTEM_READY|SYSTEM_STATE"
+SERVICE_STATUS = 'service_status'
+APP_READY_STATUS = 'app_ready_status'
+FAIL_REASON = 'fail_reason'
+UPDATE_TIME = 'update_time'
+
+class SysreadyShow(object):
+    def __init__(self):
+        self.db = SonicV2Connector(host="127.0.0.1")
+        self.db.connect(self.db.STATE_DB)
+
+    def show(self, cmd):
+        keys = self.db.keys(self.db.STATE_DB, SERVICE_STATUS_TABLE + '*')
+        if not keys:
+            print('No system ready status data available - system-health service might be down\n')
+            return
+
+        if cmd == "alldetail":
+            header_info = header_detail
+        else:
+            header_info = header
+
+        table = []
+        for key in natsorted(keys):
+            key_list = key.split('|')
+            if len(key_list) != 2: # error data in DB, log it and ignore
+                print('Warn: Invalid key in table {}: {}'.format(SERVICE_STATUS_TABLE, key))
+                continue
+
+            name = key_list[1]
+            data_dict = self.db.get_all(self.db.STATE_DB, key)
+            try:
+                service_status = data_dict[SERVICE_STATUS]
+                app_ready_status = data_dict[APP_READY_STATUS]
+                fail_reason = data_dict[FAIL_REASON]
+                update_time = data_dict[UPDATE_TIME]
+            except ValueError as e:
+                print('Error in data_dict')
+
+            if cmd == "alldetail":
+                table.append((name, service_status, app_ready_status, fail_reason, update_time))
+            else:
+                table.append((name, service_status, app_ready_status, fail_reason))
+        
+        sysready_state = self.db.get(self.db.STATE_DB, SYSREADY_TABLE, "Status")
+        if sysready_state == "UP":
+            print("System is ready\n")
+        else:
+            print("System is not ready - one or more services are not up\n")
+        
+
+        if cmd == "allbrief":
+            return
+
+
+        if table:
+            print(tabulate(table, header_info, tablefmt='simple', stralign='left'))
+        else:
+            print('No sysready status data available\n')
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Display the System Ready status',
+                                     formatter_class=argparse.RawTextHelpFormatter,
+                                     epilog="""
+                                      Examples:
+                                      sysreadyshow --all
+                                      sysreadyshow --allbrief
+                                      sysreadyshow --alldetail
+                                      """)
+
+    parser.add_argument('-a', '--all', action='store_true', help='all service status', default=True)
```

Comment:
Is there a difference between giving the `-a` flag and not giving this flag? If not, can it be removed, and the other options be renamed to `--brief` and `--detail`?

Reply:
Done